src/Utils/XPath.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace RobRichards\XMLSecLibs\Utils;
+
+class XPath
+{
+    const ALPHANUMERIC = '[^\w\d]';
+    const NUMERIC = '[^\d]';
+    const LETTERS = '[^\w]';
+    const EXTENDED_ALPHANUMERIC = '[^\w\d\s-_:\.]';
+
+    const SINGLE_QUOTE = '\'';
+    const DOUBLE_QUOTE = '"';
+    const ALL_QUOTES = '[\'"]';
+
+
+    /**
+     * Filter an attribute value for save inclusion in an XPath query.
+     *
+     * @param string $value The value to filter.
+     * @param string $quotes The quotes used to delimit the value in the XPath query.
+     *
+     * @return string The filtered attribute value.
+     */
+    public static function filterAttrValue($value, $quotes = self::ALL_QUOTES)
+    {
+        return preg_replace('#'.$quotes.'#', '', $value);
+    }
+
+
+    /**
+     * Filter an attribute name for save inclusion in an XPath query.
+     *
+     * @param string $name The attribute name to filter.
+     * @param mixed $allow The set of characters to allow. Can be one of the constants provided by this class, or a
+     * custom regex without the '#' character (used as delimiter).
+     *
+     * @return string The filtered attribute name.
+     */
+    public static function filterAttrName($name, $allow = self::EXTENDED_ALPHANUMERIC)
```

Comment:
I would just change the name of allow here as its actually the ones to filter out rather than allow.

Reply:
Actually, the name was right, but the implementation not very safe for custom sets of characters. The problem is that if you pass a custom string, it's unlikely that you include the `[^` and `]` to negate the selection, as the documentation wasn't explicitly saying that was a requirement. So I've changed the default regular expressions and move the negation to the `preg_replace()` call. That way you can pass just a custom set of allowed characters and get only those in the resulting string.